### ISSUE TITLE
Porter go library creates out of memory situation upon file system read actions

### ISSUE THREAD

**Issue by Ka0o0** [getporter/porter#1557]
## Describe the bug
I didn't manage to use the Go library in its current version as it will always consume my computers memory and make it crash. I've created a simple test case where I try to list all installations.

When debugging, it starts to hang during the connect process of the rpc client (source file [source](https://github.com/getporter/porter/blob/v0.38.1/pkg/plugins/pluggable/loader.go#L112)). Any suggestions what I'm doing wrong?

## To Reproduce
1.Open the example and step through the code in src/main.go
2. It will hang on line 11 where it tries to list all installations

## Expected behavior
that all installations are returned by the method and no out of memory situation occurs.
 
## Example
[example](https://github.com/getporter/porter/files/6427337/porter-claims.zip)

## Version
Copy the output of `porter version` below
porter v0.38.1 (3018b91c)
Fedora 34
Driver: moby

**Comment by carolynvs** [comment]
Hey @Ka0o0! It's super cool to see someone using Porter's libraries. Please be a bit patient as not many people have tried this yet and our documentation doesn't cover this very much.

I have edited your main.go to get listing installations to work (below). What was happening is that the porter library was attempting to connect to a plugin (we use plugins for storage, so that people can write to the file system or cloud storage).

This is the important line, where we tell Porter where to locate the default plugins (which are compiled into the porter binary).

```
p.SetPorterPath(filepath.Join(os.Getenv("PORTER_HOME"), "porter"))
```

```go
package main

import (
	"log"
	"os"
	"path/filepath"

	"get.porter.sh/porter/pkg/porter"
	"github.com/davecgh/go-spew/spew"
)

func main() {
	p := porter.New()

	// The filesystem storage plugin is inside the porter binary
	// PorterPath defaults to the calling binary. Since we aren't porter
	// Tell it where to find porter.
	p.SetPorterPath(filepath.Join(os.Getenv("PORTER_HOME"), "porter"))

	// Print extra details to stderr, notably that path of the plugin
	// porter is communicating with
	p.Debug = true
	p.DebugPlugins = true

	c, e := p.ListInstallations()
	if e != nil {
		log.Fatal(e)
	}

	spew.Dump(c)
}
```

```console 
$ PORTER_HOME=~/.porter go run src/main.go
Resolved storage plugin to storage.porter.filesystem
/Users/carolynvs/.porter/porter plugin run storage.porter.filesystem
(porter.DisplayInstallations) (len=3 cap=4) {
...
}
```

I have a question for you. Is your plan to use the porter library on a machine that already has porter installed? Or that will work with a specific plugin? Or do you want it to work off of Porter's configuration, where the user can use any arbitrary plugin? 

I'm asking because we haven't designed much around people using porter as a library, and it would help to understand how you wanted to use porter. Any feedback you provide can help make this a better experience for you and others!

**Comment by carolynvs** [comment]
@Ka0o0 I hope the code above got you past the problem that you were seeing. Please let us know if there are further difficulties with using the library! I would love to help you build on top of Porter. 👍

**Comment by Ka0o0** [comment]
Hi @carolynvs,

thanks for getting back. I haven't had the chance to try it out yet as I had to shift my focus to some other tasks. Regarding your suggested fix, it makes sense to me. Actually, I was thinking that the library would work a bit different and I guess this is what you were asking for with your question? So I thought that the library would simply be a wrapper around Go's exec functionality which provides some structs to help writing some code around it. 

Basically, I thought that the library would simply reach out to the locally installed/available porter binary. In the long run though, especially considering when porter would run as an K8s operator I think I would expect that the library would perform some RPCs to a running porter instance. Similarly to what kubectl is doing. What do you think?

Please feel free to close this issue.

**Comment by carolynvs** [comment]
Oh! I see what you mean. Thanks for sharing what you expected the library to do.

Yeah, the porter library right now is the underlying go library that we used to build the porter CLI. In the future as we develop the Porter Operator, the porter CLI (and this library) will also be able to communicate remotely with that operator, much more in the fashion you were expecting, like kubectl.

I'll keep this in mind as I work on the operator and document how to use the CLI with it. Thanks!
